core/internal/runfiles/saved_file.go

```diff
@@ -99,13 +99,14 @@ func (f *savedFile) Upload(
 //
 // It must be called while a lock is held. It temporarily releases the lock.
 func (f *savedFile) doUpload(uploadURL string, uploadHeaders []string) {
-	task := &filetransfer.Task{
-		FileKind: f.category,
-		Type:     filetransfer.UploadTask,
-		Path:     f.realPath,
-		Name:     string(f.runPath),
-		Url:      uploadURL,
-		Headers:  uploadHeaders,
+	task := &filetransfer.DefaultUploadTask{
+		DefaultTask: filetransfer.DefaultTask{
```

Comment:
If `DefaultUploadTask` is declared using

```
type DefaultUploadTask DefaultTask
```

then you don't need this extra layer of nesting.